lib/src/r_tree/leaf_node.dart

```diff
@@ -29,12 +29,12 @@ class LeafNode<E> extends Node<E> {
   }
 
   Iterable<RTreeDatum<E>> search(
-      Rectangle searchRect, bool Function(E item) shouldInclude) {
-    return _items.where((RTreeDatum<E> item) =>
-        item.overlaps(searchRect) && shouldInclude(item.value));
+      Rectangle? searchRect, bool Function(E item) shouldInclude) {
```

Comment:
Is `searchRect` nullable only because it's public?
Why isn't `shouldInclude` nullable?